pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/DefaultAggregationExecutor.java

```diff
@@ -179,7 +180,7 @@ private void aggregateColumn(AggregationFunctionContext aggrFuncContext, Aggrega
       case AggregationFunctionFactory.PERCENTILEEST95_MV_AGGREGATION_FUNCTION:
       case AggregationFunctionFactory.PERCENTILEEST99_MV_AGGREGATION_FUNCTION:
         aggregationFunction.aggregate(length, resultHolder,
-            (Object) _dataBlockCache.getDoubleValuesArrayForColumn(aggrColumn));
+            (Object) blockValueSet.getMultiValues());
```

Comment:
Add a TODO here to remove this in the future. This will allocate Object per value base.